cpp/src/Slice/Scanner.cpp

```diff
@@ -1970,28 +1970,28 @@ case 28:
 YY_RULE_SETUP
 #line 362 "src/Slice/Scanner.l"
 {
-    popState();
-
     string comment(yytext);
     // The last 2 characters are the '*/' matched by this rule.
     if (YY_START == C_DOC_COMMENT)
     {
         currentUnit->setDocComment(comment.substr(0, yyleng - 2));
     }
+
+    popState();
```

Comment:
The issue was that we were popping off the current state before we checked it.

When this function is called, the scanner is in `C_DOC_COMMENT` mode.
But calling `popState` changes the scanner to it's previous state.
Meaning `YY_START` (the current state) would not be what we expect here.